runtime/src/bank.rs

```diff
@@ -3351,17 +3351,24 @@ impl Bank {
         ))
     }
 
+    /// Attempt to take locks on the accounts in a transaction batch
+    pub fn try_lock_accounts(&self, txs: &[SanitizedTransaction]) -> Vec<Result<()>> {
+        let tx_account_lock_limit = self.get_transaction_account_lock_limit();
+        self.rc
+            .accounts
+            .lock_accounts(txs.iter(), tx_account_lock_limit)
+    }
+
     /// Prepare a locked transaction batch from a list of sanitized transactions.
     pub fn prepare_sanitized_batch<'a, 'b>(
         &'a self,
         txs: &'b [SanitizedTransaction],
     ) -> TransactionBatch<'a, 'b, SanitizedTransaction> {
-        let tx_account_lock_limit = self.get_transaction_account_lock_limit();
-        let lock_results = self
-            .rc
-            .accounts
-            .lock_accounts(txs.iter(), tx_account_lock_limit);
-        TransactionBatch::new(lock_results, self, OwnedOrBorrowed::Borrowed(txs))
+        TransactionBatch::new(
+            self.try_lock_accounts(txs),
+            self,
+            OwnedOrBorrowed::Borrowed(txs),
+        )
```

Comment:
just want to confirm my understanding. but this change isn't strictly needed. just a refactor?

Reply:
oh, it looks like `try_lock_accounts` is `pub`-ed and called by blockstore_processor...

Reply:
Yeah pure refactor, and I'll explain my thinking:

1. I wanted a convenient way to lock transactions in blockstore_processor (without reaching through `rc.accounts`!)
2. There's already an equivalent unlocking version
3. Since there's a new lock function, might as well use it to reduce the code dup in the original `prepare_sanitized_batch` fn. Compiler will possibly inline it anyway